_posts/2021-05-22-fluxv2-openfaas.md

```diff
@@ -0,0 +1,332 @@
+to provide us with our cluster. 
+
+You can also use `kind`, `k3d` or `minikube` to provision a cluster locally too!
+
+### Kubernetes on Linode
+1. Create an account on Linode, [this is the signup page](https://login.linode.com/signup). New users get $100 credit (at time of writing)
+
+2. Head to the `kubernetes` tab on the sidebar
+
+3. Create a cluster in a region close to you and select at least 3 nodes with 1cpu and 2GB ram. This should be enough for 
+our applications.
+   
+4. Download the `kubeconfig` once the cluster has been created and set the `KUBECONFIG` environment variable
+
+In my case, this is how to set the environment variable.
+```sh 
+export KUBECONFIG=~/Downloads/kubeconfig
+```
+
+### Kubernetes locally with `kind`
+
+There is nothing stopping you using a local cluster on your laptop. With just a few commands you can spin up a local cluster instead. 
+You wont be able to expose your functions on the internet without some extra software like [inlets](https://inlets.dev) to provide a public IP.
+
+Create a `kind` cluster abd wait for it to be ready.
+
+```sh
+kind create cluster \
+  --name flux-openfaas
+  
+kind export kubeconfig --name flux-openfaas 
+```
+
+# Install Flux
+Flux uses a CLI to setup and install the flux controllers into the cluster. Once we have the CLI we can 
+install the controllers and then start adding resources into our cluster! We installed the `flux1 cli` earlier, so we are ready to go.
+
+We have the Flux cli we need to `bootstrap` the cluster and create a git repository. 
+
+For this section I'm using Github, you can find instructions for other Git providers [in the flux docs](https://fluxcd.io/docs/installation/#bootstrap)
+
+Generate a `github personal access token` with `repo` permissions [like this](https://docs.github.com/en/github/authenticating-to-github/creating-a-personal-access-token)
+
+Export this as an environment variable
+
+```sh 
+export GITHUB_TOKEN=<your-token>
+export GITHUB_USER=<your github username>
+```
+
+Run the bootstrap for a repository on your personal GitHub account:
+
+```sh 
+flux bootstrap github \
+  --owner=$GITHUB_USER \
+  --repository=openfaas-flux \
+  --path=clusters/linode \
+  --personal
+```
+
+After a short time we should see the flux controllers installed into the `flux-system` namespace!
+
+```sh 
+$ kubectl get pods -n flux-system
+NAME                                       READY   STATUS    RESTARTS   AGE
+helm-controller-85bfd4959d-w5ccr           1/1     Running   0          38s
+kustomize-controller-6977b8cdd4-gblq7      1/1     Running   0          38s
+notification-controller-5c4d48f476-7tdkk   1/1     Running   0          38s
+source-controller-85fb864746-gzrxq         1/1     Running   0          38s
+```
+
+This step has:
+* Installed the `flux controllers` into your cluster 
+* Created a github repository, named `openfaas-flux` (unless you changed that). 
+* Added a `Deploy key` to Github with access to that repository
```

Comment:
Care to add a note on what a Deploy can is or link to docs about them? I hadn't encountered them until using Flux for the first time, so it may benefit the new user.